charts/consul/templates/cni-daemonset.yaml

```diff
@@ -0,0 +1,91 @@
+  namespace: {{ .Values.connectInject.cni.namespace }}
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: cni 
+spec:
+  {{- if .Values.connectInject.cni.updateStrategy }}
+  updateStrategy:
+    {{ tpl .Values.connectInject.cni.updateStrategy . | nindent 4 | trim }}
+  {{- end }}
+  selector:
+    matchLabels:
+      app: {{ template "consul.name" . }}
+      chart: {{ template "consul.chart" . }}
+      release: {{ .Release.Name }}
+      component: cni
+  template:
+    metadata:
+      labels:
+        app: {{ template "consul.name" . }}
+        chart: {{ template "consul.chart" . }}
+        release: {{ .Release.Name }}
+        component: cni
+      annotations:
+        consul.hashicorp.com/connect-inject: "false"
+    spec:
+      # consul-cni only runs on linux operating systems
+      nodeSelector:
+        kubernetes.io/os: linux
+      tolerations:
+        # Make sure consul-cni-node gets scheduled on all nodes.
+        - effect: NoSchedule
+          operator: Exists
+        # Mark the pod as a critical add-on for rescheduling.
+        - key: CriticalAddonsOnly
+          operator: Exists
+        - effect: NoExecute
+          operator: Exists
+      # Tell kubernetes that this daemonset is critical so that it will be scheduled on a new node before other pods
+      priorityClassName: system-node-critical
+      serviceAccountName: {{ template "consul.fullname" . }}-cni
+      {{- if not .Values.global.openshift.enabled }}
+      securityContext:
+        {{- toYaml .Values.connectInject.cni.securityContext | nindent 8 -}}
+      {{- end }}
+      # Minimize downtime during a rolling upgrade or deletion; tell Kubernetes to do a "force
+      # deletion": https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods.
+      terminationGracePeriodSeconds: 5
+      containers:
+        # This container installs the consul CNI binaries and CNI network config file on each node
+        - name: install-cni
+          image: {{ .Values.global.imageK8S }}
+          securityContext:
+            privileged: {{ .Values.connectInject.cni.privileged }}
+          command:
+          - "/bin/sh"
+          - "-ec"
+          - |
+            consul-k8s-control-plane install-cni \
+              -multus={{ .Values.connectInject.cni.multus }} \
+              -log-level={{ default .Values.global.logLevel .Values.connectInject.cni.logLevel }} \
+              -cni-bin-dir={{ .Values.connectInject.cni.cniBinDir }} \
+              -cni-net-dir={{ .Values.connectInject.cni.cniNetDir }} \
+              -dns-prefix={{ template "consul.fullname" . }}
+          {{- with .Values.connectInject.cni.resources }}
+          resources:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - mountPath: /host{{ .Values.connectInject.cni.cniBinDir }}
```

Comment:
why do we need a `/host` prefix?

Reply:
I thought that hostPath and mountPath had to be different. I just looked it up and they can be the same. Maybe I will switch it as it might make the code a little easier to comprehend also...

Reply:
I have now changed this and it works. It has made the code a bit simpler too!  Thanks for finding it!